docs/usage.md

```diff
@@ -96,36 +96,93 @@ Action Scheduler will later initialize itself on `'init'` with priority `1`.  Ac
 
 When using Action Scheduler in themes, it's important to note that if Action Scheduler has been registered by a plugin, then the latest version registered by a plugin will be used, rather than the version included in the theme. This is because of the version dependency handling code using `'plugins_loaded'` since version 1.0.
 
-## Scheduling an Action
+## Scheduling Recurring Actions on Activation and Ensuring They Remain Scheduled
 
-To schedule an action, call the [API function](/api/) for the desired schedule type passing in the required parameters.
+When developing a plugin or extension that relies on recurring actions, it is essential to schedule those actions when the plugin is first activated or updated.
 
-The example code below shows everything needed to schedule a function to run at midnight, if it's not already scheduled:
+### Best Practice: Schedule on Activation
 
+Plugins should always schedule any important recurring actions during activation or upgrade routines. This ensures the action is registered as soon as the plugin is enabled.
+
+**Example: Scheduling on Activation**
 ```php
-require_once( plugin_dir_path( __FILE__ ) . '/libraries/action-scheduler/action-scheduler.php' );
+/**
+ * Schedule recurring action on plugin activation.
+ */
+function my_plugin_activate() {
+    if ( ! as_has_scheduled_action( 'my_plugin_recurring_action' ) ) {
+        as_schedule_recurring_action(
+            time(),
+            HOUR_IN_SECONDS,
+            'my_plugin_recurring_action'
+        );
+    }
+}
+register_activation_hook( __FILE__, 'my_plugin_activate' );
+```
+
+### Ensuring Recurring Actions Remain Scheduled
+
+After initial scheduling, there may be rare cases where a recurring action is lost (e.g., due to a database failure or repeated action failures). The `action_scheduler_ensure_recurring_actions` hook provides a reliable opportunity to check and reschedule the action if necessary. 
 
+**Example: Using the Ensure Hook**
+```php
 /**
- * Schedule an action with the hook 'eg_midnight_log' to run at midnight each day
- * so that our callback is run then.
+ * Ensure recurring action remains scheduled.
  */
-function eg_schedule_midnight_log() {
-	if ( false === as_has_scheduled_action( 'eg_midnight_log' ) ) {
-		as_schedule_recurring_action( strtotime( 'tomorrow' ), DAY_IN_SECONDS, 'eg_midnight_log', array(), '', true );
+add_action( 'action_scheduler_ensure_recurring_actions', function() {
+    if ( ! as_has_scheduled_action( 'my_plugin_recurring_action' ) ) {
+        as_schedule_recurring_action(
```

Comment:
*(I realize this PR was merged already, I was just catching up on the latest changes ... nice work on this, btw!)*

This leads to a race condition, because a second process may create the same recurring action right after our check to see if it exists.

- What do you think of setting `unique` to true?
- Going off on a bit of a tangent, what do you think about adding support for `as_supports( 'unique-actions' )` (on the basis that, on those rare occasions where a plugin assumes unique actions available, but they are not, a flood of actions may be created)?